src/Infrastructure.Styles.Analyzer/RMSYN1102SeparateExitConditionsAnalyzer/RMSYN1102SeparateExitConditionsCodeFixProvider.cs

```diff
@@ -0,0 +1,180 @@
+// along with re-motion; if not, see http://www.gnu.org/licenses.
+
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Text;
+
+namespace Infrastructure.Styles.Analyzer.RMSYN1102SeparateExitConditionsAnalyzer
+{
+  public class RMSYN1102SeparateExitConditionsCodeFixProvider : CodeFixProvider
+  {
+    public override FixAllProvider? GetFixAllProvider ()
+    {
+      return WellKnownFixAllProviders.BatchFixer;
+    }
+
+    public override Task RegisterCodeFixesAsync (CodeFixContext context)
+    {
+      context.RegisterCodeFix(
+        CodeAction.Create(
+          "Split IF statement",
+          c => CodeFix(c, context),
+          RMSYN1102SeparateExitConditionsAnalyzer.DiagnosticId),
+        context.Diagnostics.First());
+      return Task.CompletedTask;
+    }
+
+    public override ImmutableArray<string> FixableDiagnosticIds { get; } =
+      ImmutableArray.Create(RMSYN1102SeparateExitConditionsAnalyzer.DiagnosticId);
+
+    private static async Task<Document> CodeFix (CancellationToken cancellationToken, CodeFixContext context)
+    {
+      var diagnostic = context.Diagnostics.First();
+      var diagnosticSpan = diagnostic.Location.SourceSpan;
+
+      var syntaxRoot = await context.Document.GetSyntaxRootAsync(cancellationToken);
+      if (syntaxRoot == null) return context.Document;
+
+      return context.Document.WithSyntaxRoot(ReplaceIllegalStatement(syntaxRoot, diagnosticSpan));
+    }
+
+    private static SyntaxNode ReplaceIllegalStatement (SyntaxNode syntaxRoot, TextSpan diagnosticSpan)
+    {
+      IllegalExpression illegalExpression;
+      try
+      {
+        illegalExpression = new IllegalExpression(syntaxRoot.FindNode(diagnosticSpan));
+      }
+      catch
+      {
+        return syntaxRoot;
+      }
+
+      var syntaxWrapper = CreateSyntaxWrapper(illegalExpression.IfStatement);
+      
+      //need to remove else otherwise there are else clauses after every if and that's not desirable
+      var syntaxWrapperWithoutElse = syntaxWrapper.Copy();
+      syntaxWrapperWithoutElse.RemoveElseClause();
+
+      var syntaxNodes = new List<IfAndElseIfStatementSyntaxWrapper>();
+
+      FillSyntaxNodeListWithStatements(syntaxNodes, syntaxWrapperWithoutElse,
+        illegalExpression.Node);
+
+      //time to handle the previously ignored else clauses
+      if (syntaxWrapper.HasElseClause)
+        HandleElseClauses(syntaxWrapper, syntaxNodes);
+
+      return BuildRoot(syntaxRoot, syntaxNodes, syntaxWrapper);
+    }
+
+    private static void HandleElseClauses (IfAndElseIfStatementSyntaxWrapper syntaxWrapper,
+      List<IfAndElseIfStatementSyntaxWrapper> syntaxNodes)
```

Comment:
```suggestion
    private static void HandleElseClauses (
      IfAndElseIfStatementSyntaxWrapper syntaxWrapper,
      List<IfAndElseIfStatementSyntaxWrapper> syntaxNodes)
```

If the parameters require a line-break all of them should be in their own line and the first one is also in a new line. Please fix the other methods as well